sycl/source/kernel_bundle.cpp

```diff
@@ -294,5 +295,38 @@ std::vector<kernel_id> get_kernel_ids() {
   return detail::ProgramManager::getInstance().getAllSYCLKernelIDs();
 }
 
+bool is_compatible(const std::vector<kernel_id> &KernelIDs, const device &Dev) {
+  if (KernelIDs.empty())
+    return false;
+  for (const auto &KernelId : KernelIDs) {
+    const detail::RTDeviceBinaryImage &Img =
+        detail::ProgramManager::getInstance().getDeviceImage(
+            detail::OSUtil::ExeModuleHandle, KernelId.get_name(), context(Dev),
+            Dev);
+    const detail::RTDeviceBinaryImage::PropertyRange &ARange =
+        Img.getDeviceRequirements();
+    for (detail::RTDeviceBinaryImage::PropertyRange::ConstIterator It :
+         ARange) {
+      using namespace std::literals;
+      if ((*It)->Name != "aspects"sv)
+        continue;
+      detail::ByteArray Aspects =
+          detail::DeviceBinaryProperty(*It).asByteArray();
+      // 8 because we need to skip 64-bits of size of the byte array
+      auto *AIt = reinterpret_cast<const std::uint32_t *>(&Aspects[8]);
+      auto *AEnd =
+          reinterpret_cast<const std::uint32_t *>(&Aspects[0] + Aspects.size());
+      while (AIt != AEnd) {
+        auto Aspect = static_cast<aspect>(*AIt);
```

Comment:
Likewise, https://github.com/intel/llvm/blob/7e0f76bba6844aa67d8a230463cf53f2581ab92b/sycl/source/detail/device_binary_image.hpp#L32 seems to handle this.

Reply:
Yep, thanks [1e9ea5e](https://github.com/intel/llvm/pull/7510/commits/1e9ea5e091ae2df93aefebd837b616b54fdfbe54)